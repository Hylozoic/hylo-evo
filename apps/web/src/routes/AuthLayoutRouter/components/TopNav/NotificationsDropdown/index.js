import component from './NotificationsDropdown'
import connector from './NotificationsDropdown.connector'

export default connector(component)
