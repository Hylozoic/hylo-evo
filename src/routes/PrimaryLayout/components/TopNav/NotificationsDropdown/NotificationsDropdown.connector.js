import { connect } from 'react-redux'
import {
  fetchNotifications,
  markActivityRead,
  markAllActivitiesRead,
  getNotifications,
  goToNotification
} from './NotificationsDropdown.store'
import getMe from 'store/selectors/getMe'
import { FETCH_NOTIFICATIONS } from 'store/constants'

export function mapStateToProps (state, props) {
  const notifications = getNotifications(state, props)
  return {
    notifications,
    currentUser: getMe(state, props),
    pending: state.pending[FETCH_NOTIFICATIONS]
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    fetchNotifications: () => dispatch(fetchNotifications()),
    goToNotification: notification => dispatch(goToNotification(notification)),
    markActivityRead: id => dispatch(markActivityRead(id)),
    markAllActivitiesRead: () => dispatch(markAllActivitiesRead())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
