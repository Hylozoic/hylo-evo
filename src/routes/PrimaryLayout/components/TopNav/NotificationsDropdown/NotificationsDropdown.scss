// NotificationsDropdown

.header-content {
  composes: header-content from '../TopNavDropdown/TopNavDropdown.scss';
}

.tab {
  composes: header-link from '../TopNavDropdown/TopNavDropdown.scss';
  margin-right: 15px;
}

.active {
  color: $color-rhino;
  border-bottom: 2px solid $color-rhino;
  position: relative;
  top: 4px;
  padding: 0 4px 7px 4px;
  margin: 0 11px 0 -4px;
  &:hover {
    color: $color-rhino-80;
    border-bottom: 2px solid $color-rhino-80;
  }
}

.mark-read {
  composes: header-link from '../TopNavDropdown/TopNavDropdown.scss';
  margin-left: auto;
}

.notifications {
  composes: body from '../TopNavDropdown/TopNavDropdown.scss';
}

.notification {
  display: flex;
  align-items: start;
  cursor: pointer;
  &:first-child {
    margin-top: 5px;
  }
  &:last-child {
    .message-content {
      border: none;
    }
  }
}

.unread {
  background-color: $color-saltpan;
}

.image-wraper {
  margin: 0 10px 0 15px;
  padding-top: 20px;
}

.date {
  composes: caption-lt-lg from 'css/typography.scss';
  color: $color-slate-gray-80;
  font-size: 12px;
}

.content {
  display: flex;
  flex-direction: column;
  align-items: start;
  padding: 20px 15px 15px 0;
  border-bottom: 1px solid $color-black-squeeze;
}

// NotificationHeader

.header {
  composes: caption-lt-lg from 'css/typography.scss';
  margin-bottom: 5px;
  color: $color-rhino;
}

.bold {
  @include font-bold;
}

// NotificationBody

.body {
  composes: caption-lt-lg from 'css/typography.scss';
  color: $color-rhino-60;
}
