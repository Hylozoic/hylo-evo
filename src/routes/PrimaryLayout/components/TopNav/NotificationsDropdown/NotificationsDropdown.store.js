import { FETCH_NOTIFICATIONS } from 'store/constants'
import { createSelector as ormCreateSelector } from 'redux-orm'
import orm from 'store/models'
import { push } from 'react-router-redux'
import {
  ACTION_NEW_COMMENT,
  ACTION_TAG,
  ACTION_JOIN_REQUEST,
  ACTION_APPROVED_JOIN_REQUEST,
  ACTION_MENTION,
  ACTION_COMMENT_MENTION
} from 'store/models/Notification'
import {
  commentUrl,
  postUrl,
  communityUrl,
  communitySettingsUrl
} from 'util/index'

export function fetchNotifications () {
  return {
    type: FETCH_NOTIFICATIONS,
    graphql: {
      query: `{
        me {
          name
          notifications (first: 10, order: "desc") {
            total
            hasMore
            items {
              id
              createdAt
              activity {
                id
                actor {
                  id
                  name
                  avatarUrl
                }
                comment {
                  id
                  text
                }
                post {
                  id
                  title
                }
                community {
                  id
                  name
                  slug
                }
                meta {
                  reasons
                }
                action
                unread
              }
            }
          }
        }
      }`
    },
    meta: {
      extractModel: 'Me'
    }
  }
}

export function urlForNotification ({ activity: { action, post, comment, community } }) {
  switch (action) {
    case ACTION_NEW_COMMENT:
    case ACTION_COMMENT_MENTION:
      return commentUrl(post.id, comment.id)
    case ACTION_TAG:
    case ACTION_MENTION:
      return postUrl(post.id, comment.id)
    case ACTION_JOIN_REQUEST:
      return communitySettingsUrl(community.slug)
    case ACTION_APPROVED_JOIN_REQUEST:
      return communityUrl(community.slug)
  }
}

export function goToNotification (notification) {
  return push(urlForNotification(notification))
}

export const getNotifications = ormCreateSelector(
  orm,
  state => state.orm,
  (session) => {
    return session.Notification.all().toModelArray()
  }
)
