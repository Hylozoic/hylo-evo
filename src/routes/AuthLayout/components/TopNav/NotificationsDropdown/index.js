import NotificationsDropdown from './NotificationsDropdown'
import connector from './NotificationsDropdown.connector'
export default connector(NotificationsDropdown)
