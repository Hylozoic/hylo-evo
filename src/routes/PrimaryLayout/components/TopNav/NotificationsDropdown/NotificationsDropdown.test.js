import NotificationsDropdown, { Notification, NotificationHeader, NotificationBody } from './NotificationsDropdown'
import { shallow } from 'enzyme'
import React from 'react'
import {
  ACTION_NEW_COMMENT,
  ACTION_TAG,
  ACTION_JOIN_REQUEST,
  ACTION_APPROVED_JOIN_REQUEST,
  ACTION_MENTION,
  ACTION_COMMENT_MENTION
 } from 'store/models/Notification'

const u1 = {id: 1, name: 'Charles Darwin', avatarUrl: 'foo.png'}
const u2 = {id: 2, name: 'Marie Curie', avatarUrl: 'bar.png'}
const u3 = {id: 3, name: 'Arthur Fonzarelli', avatarUrl: 'baz.png'}

const commentNotification = {
  id: 1,
  activity: {
    actor: u2,
    action: ACTION_NEW_COMMENT,
    meta: {},
    post: {title: 'Our Oceans'},
    comment: {
      text: "I live right next to there and can come help out. I've never done petitioning but I'm sure it's an absolute blast"
    },
    unread: true
  },
  createdAt: new Date(Date.UTC(1995, 11, 17, 3, 23, 0))
}

const tagNotification = {
  id: 2,
  activity: {
    actor: u3,
    action: ACTION_TAG,
    post: {title: 'I have so many things I need!'},
    meta: {reasons: ['tag: request']},
    unread: true
  },
  createdAt: new Date(Date.UTC(1995, 11, 17, 3, 23, 0))
}

const joinRequestNotification = {
  id: 3,
  activity: {
    actor: u2,
    action: ACTION_JOIN_REQUEST,
    meta: {},
    community: {name: 'Foomunity'},
    unread: true
  },
  createdAt: new Date(Date.UTC(1995, 11, 17, 3, 23, 0))
}

const approvedJoinRequestNotification = {
  id: 4,
  activity: {
    actor: u2,
    action: ACTION_APPROVED_JOIN_REQUEST,
    meta: {},
    community: {name: 'Foomunity'},
    unread: true
  },
  createdAt: new Date(Date.UTC(1995, 11, 17, 3, 23, 0))
}

const mentionNotification = {
  id: 5,
  activity: {
    actor: u2,
    action: ACTION_MENTION,
    meta: {},
    post: {title: 'Heads up'},
    unread: true
  },
  createdAt: new Date(Date.UTC(1995, 11, 17, 3, 23, 0))
}

const commentMentionNotification = {
  id: 1,
  activity: {
    actor: u2,
    action: ACTION_COMMENT_MENTION,
    meta: {},
    post: {title: 'Our Oceans'},
    comment: {
      text: "I live right next to there and can come help out. I've never done petitioning but I'm sure it's an absolute blast"
    },
    unread: true
  },
  createdAt: new Date(Date.UTC(1995, 11, 17, 3, 23, 0))
}

const notifications = [
  commentNotification,
  tagNotification,
  {...commentNotification, unread: false},
  {...tagNotification, unread: false},
  joinRequestNotification,
  approvedJoinRequestNotification,
  mentionNotification,
  commentMentionNotification
]

describe('NotificationsDropdown', () => {
  it('renders correctly with an empty list', () => {
    const wrapper = shallow(<NotificationsDropdown
      renderToggleChildren={() => <span>click me</span>}
      notifications={[]}
      currentUser={u1} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a list of threads', () => {
    const wrapper = shallow(<NotificationsDropdown
      renderToggleChildren={() => <span>click me</span>}
      notifications={notifications}
      currentUser={u1} />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Notification', () => {
  it('renders correctly with a comment notification', () => {
    const wrapper = shallow(<Notification notification={commentNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a tag notification', () => {
    const wrapper = shallow(<Notification notification={tagNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a join request notification', () => {
    const wrapper = shallow(<Notification notification={joinRequestNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with an approved join request notification', () => {
    const wrapper = shallow(<Notification notification={approvedJoinRequestNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a mention notification', () => {
    const wrapper = shallow(<Notification notification={mentionNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a comment mention notification', () => {
    const wrapper = shallow(<Notification notification={commentMentionNotification} />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('NotificationHeader', () => {
  it('renders correctly with a comment notification', () => {
    const wrapper = shallow(<NotificationHeader notification={commentNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a tag notification', () => {
    const wrapper = shallow(<NotificationHeader notification={tagNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a join request notification', () => {
    const wrapper = shallow(<NotificationHeader notification={joinRequestNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with an approved join request notification', () => {
    const wrapper = shallow(<NotificationHeader notification={approvedJoinRequestNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a mention notification', () => {
    const wrapper = shallow(<NotificationHeader notification={mentionNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a comment mention notification', () => {
    const wrapper = shallow(<NotificationHeader notification={commentMentionNotification} />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('NotificationBody', () => {
  it('renders correctly with a comment notification', () => {
    const wrapper = shallow(<NotificationBody notification={commentNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a tag notification', () => {
    const wrapper = shallow(<NotificationBody notification={tagNotification} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders correctly with a join request notification', () => {
    const wrapper = shallow(<NotificationBody notification={joinRequestNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with an approved join request notification', () => {
    const wrapper = shallow(<NotificationBody notification={approvedJoinRequestNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a mention notification', () => {
    const wrapper = shallow(<NotificationBody notification={mentionNotification} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with a comment mention notification', () => {
    const wrapper = shallow(<NotificationBody notification={commentMentionNotification} />)
    expect(wrapper).toMatchSnapshot()
  })
})
