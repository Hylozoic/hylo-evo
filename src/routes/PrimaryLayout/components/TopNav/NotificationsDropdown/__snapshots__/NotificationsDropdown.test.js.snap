// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly with a comment mention notification 1`] = `
<li
  data-stylename="notification unread"
  onClick={[Function]}
>
  <div
    data-stylename="image-wraper"
  >
    <RoundImage
      url="bar.png"
    />
  </div>
  <div
    data-stylename="content"
  >
    <NotificationHeader
      notification={
        Object {
          "activity": Object {
            "action": "commentMention",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "comment": Object {
              "text": "I live right next to there and can come help out. I've never done petitioning but I'm sure it's an absolute blast",
            },
            "meta": Object {},
            "post": Object {
              "title": "Our Oceans",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 1,
        }
      }
    />
    <NotificationBody
      notification={
        Object {
          "activity": Object {
            "action": "commentMention",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "comment": Object {
              "text": "I live right next to there and can come help out. I've never done petitioning but I'm sure it's an absolute blast",
            },
            "meta": Object {},
            "post": Object {
              "title": "Our Oceans",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 1,
        }
      }
    />
    <div
      data-stylename="date"
    >
      23 years ago
    </div>
  </div>
</li>
`;

exports[`Notification renders correctly with a comment notification 1`] = `
<li
  data-stylename="notification unread"
  onClick={[Function]}
>
  <div
    data-stylename="image-wraper"
  >
    <RoundImage
      url="bar.png"
    />
  </div>
  <div
    data-stylename="content"
  >
    <NotificationHeader
      notification={
        Object {
          "activity": Object {
            "action": "newComment",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "comment": Object {
              "text": "I live right next to there and can come help out. I've never done petitioning but I'm sure it's an absolute blast",
            },
            "meta": Object {},
            "post": Object {
              "title": "Our Oceans",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 1,
        }
      }
    />
    <NotificationBody
      notification={
        Object {
          "activity": Object {
            "action": "newComment",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "comment": Object {
              "text": "I live right next to there and can come help out. I've never done petitioning but I'm sure it's an absolute blast",
            },
            "meta": Object {},
            "post": Object {
              "title": "Our Oceans",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 1,
        }
      }
    />
    <div
      data-stylename="date"
    >
      23 years ago
    </div>
  </div>
</li>
`;

exports[`Notification renders correctly with a donation to notification 1`] = `
<li
  data-stylename="notification unread"
  onClick={[Function]}
>
  <div
    data-stylename="image-wraper"
  >
    <RoundImage
      url="bar.png"
    />
  </div>
  <div
    data-stylename="content"
  >
    <NotificationHeader
      notification={
        Object {
          "activity": Object {
            "action": "donation to",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "contributionAmount": 12300,
            "meta": Object {},
            "post": Object {
              "title": "Our Oceans",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 1,
        }
      }
    />
    <NotificationBody
      notification={
        Object {
          "activity": Object {
            "action": "donation to",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "contributionAmount": 12300,
            "meta": Object {},
            "post": Object {
              "title": "Our Oceans",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 1,
        }
      }
    />
    <div
      data-stylename="date"
    >
      23 years ago
    </div>
  </div>
</li>
`;

exports[`Notification renders correctly with a donation to notification 2`] = `
<li
  data-stylename="notification unread"
  onClick={[Function]}
>
  <div
    data-stylename="image-wraper"
  >
    <RoundImage
      url="bar.png"
    />
  </div>
  <div
    data-stylename="content"
  >
    <NotificationHeader
      notification={
        Object {
          "activity": Object {
            "action": "donation from",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "contributionAmount": 12300,
            "meta": Object {},
            "post": Object {
              "title": "Our Oceans",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 1,
        }
      }
    />
    <NotificationBody
      notification={
        Object {
          "activity": Object {
            "action": "donation from",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "contributionAmount": 12300,
            "meta": Object {},
            "post": Object {
              "title": "Our Oceans",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 1,
        }
      }
    />
    <div
      data-stylename="date"
    >
      23 years ago
    </div>
  </div>
</li>
`;

exports[`Notification renders correctly with a join request notification 1`] = `
<li
  data-stylename="notification unread"
  onClick={[Function]}
>
  <div
    data-stylename="image-wraper"
  >
    <RoundImage
      url="bar.png"
    />
  </div>
  <div
    data-stylename="content"
  >
    <NotificationHeader
      notification={
        Object {
          "activity": Object {
            "action": "joinRequest",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "community": Object {
              "name": "Foomunity",
            },
            "meta": Object {},
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 3,
        }
      }
    />
    <NotificationBody
      notification={
        Object {
          "activity": Object {
            "action": "joinRequest",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "community": Object {
              "name": "Foomunity",
            },
            "meta": Object {},
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 3,
        }
      }
    />
    <div
      data-stylename="date"
    >
      23 years ago
    </div>
  </div>
</li>
`;

exports[`Notification renders correctly with a mention notification 1`] = `
<li
  data-stylename="notification unread"
  onClick={[Function]}
>
  <div
    data-stylename="image-wraper"
  >
    <RoundImage
      url="bar.png"
    />
  </div>
  <div
    data-stylename="content"
  >
    <NotificationHeader
      notification={
        Object {
          "activity": Object {
            "action": "mention",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "meta": Object {},
            "post": Object {
              "title": "Heads up",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 5,
        }
      }
    />
    <NotificationBody
      notification={
        Object {
          "activity": Object {
            "action": "mention",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "meta": Object {},
            "post": Object {
              "title": "Heads up",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 5,
        }
      }
    />
    <div
      data-stylename="date"
    >
      23 years ago
    </div>
  </div>
</li>
`;

exports[`Notification renders correctly with a tag notification 1`] = `
<li
  data-stylename="notification unread"
  onClick={[Function]}
>
  <div
    data-stylename="image-wraper"
  >
    <RoundImage
      url="baz.png"
    />
  </div>
  <div
    data-stylename="content"
  >
    <NotificationHeader
      notification={
        Object {
          "activity": Object {
            "action": "tag",
            "actor": Object {
              "avatarUrl": "baz.png",
              "id": 3,
              "name": "Arthur Fonzarelli",
            },
            "meta": Object {
              "reasons": Array [
                "tag: request",
              ],
            },
            "post": Object {
              "title": "I have so many things I need!",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 2,
        }
      }
    />
    <NotificationBody
      notification={
        Object {
          "activity": Object {
            "action": "tag",
            "actor": Object {
              "avatarUrl": "baz.png",
              "id": 3,
              "name": "Arthur Fonzarelli",
            },
            "meta": Object {
              "reasons": Array [
                "tag: request",
              ],
            },
            "post": Object {
              "title": "I have so many things I need!",
            },
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 2,
        }
      }
    />
    <div
      data-stylename="date"
    >
      23 years ago
    </div>
  </div>
</li>
`;

exports[`Notification renders correctly with an approved join request notification 1`] = `
<li
  data-stylename="notification unread"
  onClick={[Function]}
>
  <div
    data-stylename="image-wraper"
  >
    <RoundImage
      url="bar.png"
    />
  </div>
  <div
    data-stylename="content"
  >
    <NotificationHeader
      notification={
        Object {
          "activity": Object {
            "action": "approvedJoinRequest",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "community": Object {
              "name": "Foomunity",
            },
            "meta": Object {},
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 4,
        }
      }
    />
    <NotificationBody
      notification={
        Object {
          "activity": Object {
            "action": "approvedJoinRequest",
            "actor": Object {
              "avatarUrl": "bar.png",
              "id": 2,
              "name": "Marie Curie",
            },
            "community": Object {
              "name": "Foomunity",
            },
            "meta": Object {},
            "unread": true,
          },
          "createdAt": 1995-12-17T03:23:00.000Z,
          "id": 4,
        }
      }
    />
    <div
      data-stylename="date"
    >
      23 years ago
    </div>
  </div>
</li>
`;

exports[`NotificationBody renders correctly with a comment mention notification 1`] = `
<div
  data-stylename="body"
>
  <span
    data-stylename="bold"
  >
    Marie
  </span>
   wrote: "
  I live right next to there and can come help out. I've never done …
  "
</div>
`;

exports[`NotificationBody renders correctly with a comment notification 1`] = `
<div
  data-stylename="body"
>
  <span
    data-stylename="bold"
  >
    Marie
  </span>
   wrote: "
  I live right next to there and can come help out. I've never done …
  "
</div>
`;

exports[`NotificationBody renders correctly with a donation from notification 1`] = `
<div
  data-stylename="body"
>
  <span
    data-stylename="bold"
  >
    Marie Curie
  </span>
   contributed $
  123
   to "
  Our Oceans
  "
</div>
`;

exports[`NotificationBody renders correctly with a donation to notification 1`] = `
<div
  data-stylename="body"
>
  <span
    data-stylename="bold"
  >
    You
  </span>
   contributed $
  123
   to "
  Our Oceans
  "
</div>
`;

exports[`NotificationBody renders correctly with a join request notification 1`] = `
<div
  data-stylename="body"
>
  <span
    data-stylename="bold"
  >
    Marie Curie
     
  </span>
  asked to join
  <span
    data-stylename="bold"
  >
     
    Foomunity
  </span>
</div>
`;

exports[`NotificationBody renders correctly with a mention notification 1`] = `
<div
  data-stylename="body"
>
  <span
    data-stylename="bold"
  >
    Marie
  </span>
   wrote: "
  Heads up
  "
</div>
`;

exports[`NotificationBody renders correctly with a tag notification 1`] = `
<div
  data-stylename="body"
>
  <span
    data-stylename="bold"
  >
    Arthur
  </span>
   wrote: "
  I have so many things I need!
  "
</div>
`;

exports[`NotificationBody renders correctly with an announcement notification 1`] = `
<div
  data-stylename="body"
>
  <span
    data-stylename="bold"
  >
    Marie
  </span>
   wrote: "
  Announcement
  "
</div>
`;

exports[`NotificationBody renders correctly with an approved join request notification 1`] = `
<div
  data-stylename="body"
>
  <span
    data-stylename="bold"
  >
    Marie Curie
     
  </span>
  approved your request to join
  <span
    data-stylename="bold"
  >
     
    Foomunity
  </span>
</div>
`;

exports[`NotificationHeader renders correctly with a comment mention notification 1`] = `
<div
  data-stylename="header"
>
  <span
    data-stylename="bold"
  >
    Marie Curie
     
  </span>
  mentioned you in a comment on
  <span
    data-stylename="bold"
  >
     
    Our Oceans
  </span>
</div>
`;

exports[`NotificationHeader renders correctly with a comment notification 1`] = `
<div
  data-stylename="header"
>
  New Comment on 
  <span
    data-stylename="bold"
  >
    Our Oceans
  </span>
</div>
`;

exports[`NotificationHeader renders correctly with a donation from notification 1`] = `
<div
  data-stylename="header"
>
  <span
    data-stylename="bold"
  >
    Marie Curie
     
  </span>
  contributed to your project
</div>
`;

exports[`NotificationHeader renders correctly with a donation to notification 1`] = `
<div
  data-stylename="header"
>
  <span
    data-stylename="bold"
  >
    You 
  </span>
  contributed to a project
</div>
`;

exports[`NotificationHeader renders correctly with a join request notification 1`] = `
<div
  data-stylename="header"
>
  New Join Request
</div>
`;

exports[`NotificationHeader renders correctly with a mention notification 1`] = `
<div
  data-stylename="header"
>
  <span
    data-stylename="bold"
  >
    Marie Curie
     
  </span>
  mentioned you
</div>
`;

exports[`NotificationHeader renders correctly with a tag notification 1`] = `
<div
  data-stylename="header"
>
  New Post in 
  <span
    data-stylename="bold"
  >
    #
    request
  </span>
</div>
`;

exports[`NotificationHeader renders correctly with an approved join request notification 1`] = `
<div
  data-stylename="header"
>
  Join Request Approved
</div>
`;

exports[`NotificationsDropdown renders correctly with a list of threads 1`] = `
<TopNavDropdown
  body={
    <div
      data-stylename="notifications"
    >
      <Notification
        notification={
          Object {
            "activity": Object {
              "action": "newComment",
              "actor": Object {
                "avatarUrl": "bar.png",
                "id": 2,
                "name": "Marie Curie",
              },
              "comment": Object {
                "text": "I live right next to there and can come help out. I've never done petitioning but I'm sure it's an absolute blast",
              },
              "meta": Object {},
              "post": Object {
                "title": "Our Oceans",
              },
              "unread": true,
            },
            "createdAt": 1995-12-17T03:23:00.000Z,
            "id": 1,
          }
        }
        onClick={[Function]}
      />
      <Notification
        notification={
          Object {
            "activity": Object {
              "action": "tag",
              "actor": Object {
                "avatarUrl": "baz.png",
                "id": 3,
                "name": "Arthur Fonzarelli",
              },
              "meta": Object {
                "reasons": Array [
                  "tag: request",
                ],
              },
              "post": Object {
                "title": "I have so many things I need!",
              },
              "unread": true,
            },
            "createdAt": 1995-12-17T03:23:00.000Z,
            "id": 2,
          }
        }
        onClick={[Function]}
      />
      <Notification
        notification={
          Object {
            "activity": Object {
              "action": "newComment",
              "actor": Object {
                "avatarUrl": "bar.png",
                "id": 2,
                "name": "Marie Curie",
              },
              "comment": Object {
                "text": "I live right next to there and can come help out. I've never done petitioning but I'm sure it's an absolute blast",
              },
              "meta": Object {},
              "post": Object {
                "title": "Our Oceans",
              },
              "unread": true,
            },
            "createdAt": 1995-12-17T03:23:00.000Z,
            "id": 1,
            "unread": false,
          }
        }
        onClick={[Function]}
      />
      <Notification
        notification={
          Object {
            "activity": Object {
              "action": "tag",
              "actor": Object {
                "avatarUrl": "baz.png",
                "id": 3,
                "name": "Arthur Fonzarelli",
              },
              "meta": Object {
                "reasons": Array [
                  "tag: request",
                ],
              },
              "post": Object {
                "title": "I have so many things I need!",
              },
              "unread": true,
            },
            "createdAt": 1995-12-17T03:23:00.000Z,
            "id": 2,
            "unread": false,
          }
        }
        onClick={[Function]}
      />
      <Notification
        notification={
          Object {
            "activity": Object {
              "action": "joinRequest",
              "actor": Object {
                "avatarUrl": "bar.png",
                "id": 2,
                "name": "Marie Curie",
              },
              "community": Object {
                "name": "Foomunity",
              },
              "meta": Object {},
              "unread": true,
            },
            "createdAt": 1995-12-17T03:23:00.000Z,
            "id": 3,
          }
        }
        onClick={[Function]}
      />
      <Notification
        notification={
          Object {
            "activity": Object {
              "action": "approvedJoinRequest",
              "actor": Object {
                "avatarUrl": "bar.png",
                "id": 2,
                "name": "Marie Curie",
              },
              "community": Object {
                "name": "Foomunity",
              },
              "meta": Object {},
              "unread": true,
            },
            "createdAt": 1995-12-17T03:23:00.000Z,
            "id": 4,
          }
        }
        onClick={[Function]}
      />
      <Notification
        notification={
          Object {
            "activity": Object {
              "action": "mention",
              "actor": Object {
                "avatarUrl": "bar.png",
                "id": 2,
                "name": "Marie Curie",
              },
              "meta": Object {},
              "post": Object {
                "title": "Heads up",
              },
              "unread": true,
            },
            "createdAt": 1995-12-17T03:23:00.000Z,
            "id": 5,
          }
        }
        onClick={[Function]}
      />
      <Notification
        notification={
          Object {
            "activity": Object {
              "action": "commentMention",
              "actor": Object {
                "avatarUrl": "bar.png",
                "id": 2,
                "name": "Marie Curie",
              },
              "comment": Object {
                "text": "I live right next to there and can come help out. I've never done petitioning but I'm sure it's an absolute blast",
              },
              "meta": Object {},
              "post": Object {
                "title": "Our Oceans",
              },
              "unread": true,
            },
            "createdAt": 1995-12-17T03:23:00.000Z,
            "id": 1,
          }
        }
        onClick={[Function]}
      />
    </div>
  }
  header={
    <div
      data-stylename="header-content"
    >
      <span
        data-stylename="tab active"
        onClick={[Function]}
      >
        Recent
      </span>
      <span
        data-stylename="tab"
        onClick={[Function]}
      >
        Unread
      </span>
      <span
        data-stylename="mark-read"
      >
        Mark all as read
      </span>
    </div>
  }
  onToggle={[Function]}
  toggleChildren={
    <span>
      click me
    </span>
  }
/>
`;

exports[`NotificationsDropdown renders correctly with an empty list 1`] = `
<TopNavDropdown
  body={
    <NoItems
      message="No notifications"
    />
  }
  header={
    <div
      data-stylename="header-content"
    >
      <span
        data-stylename="tab active"
        onClick={[Function]}
      >
        Recent
      </span>
      <span
        data-stylename="tab"
        onClick={[Function]}
      >
        Unread
      </span>
      <span
        data-stylename="mark-read"
      >
        Mark all as read
      </span>
    </div>
  }
  onToggle={[Function]}
  toggleChildren={
    <span>
      click me
    </span>
  }
/>
`;
